c++/src/kj/compat/tls.c++

```diff
@@ -445,8 +459,58 @@ public:
   }
 
 private:
+  Promise<void> doOneAccept() {
+    // Accept one connection and queue up the next accept on our TaskSet
+
+    auto loopAgainOnSuccess = [this](AuthenticatedStream&& stream) {
+      auto promise = kj::evalNow([&] {
+        // Do the SSL acceptance procedure.
+        return tls.wrapServer(kj::mv(stream));
+      });
+
+      auto pushStream = [this](AuthenticatedStream&& stream) -> Promise<void> {
+        // This is only attached to the success path, thus the error handler will catch if our
+        // promise fails.
+        queue.push(kj::mv(stream));
+        return kj::READY_NOW;
+      };
+      tasks.add(promise.then(kj::mv(pushStream)));
+
+      // Queue up the next accept loop immediately without waiting for SSL_accept()/wrapServer().
+      tasks.add(doOneAccept());
```

Comment:
Why not `return doOneAccept();`? That's how we usually do loops like this.

(Probably `doOneAccept()` would be renamed to `acceptLoop()` in this case.)

Reply:
Hmmmm, yeah, not sure if I had a good motivation here.